app/services/video_conversion/aws_media_convert_adapter.rb

```diff
@@ -0,0 +1,266 @@
+# frozen_string_literal: true
+require "aws-sdk-mediaconvert"
+
+module VideoConversion
+  class AwsMediaConvertAdapter < BaseAdapter
+    ADAPTER_NAME = "aws_mediaconvert"
+
+    def convert
+      return false if !valid_settings?
+
+      begin
+        new_sha1 = SecureRandom.hex(20)
+        output_path = "original/1X/#{new_sha1}"
```

Comment:
It seems like we have some logic to derive the path for uploads in `FileStore::BaseStore`

https://github.com/discourse/discourse/blob/ebf709f80109e3d738a35e70371881bbb615ffe6/lib/file_store/base_store.rb#L157

Generally, I'm wondering if we can lean on the logic in `UploadCreator` here instead instead of duplicating some of the logic in this adapter.